tests/ui/consts/const-eval/issue-50814-2.mir-opt.stderr

```diff
@@ -10,6 +10,52 @@ note: erroneous constant encountered
 LL |     &<A<T> as Foo<T>>::BAR
    |      ^^^^^^^^^^^^^^^^^^^^^
 
+note: erroneous constant encountered
+  --> $DIR/issue-50814-2.rs:20:5
+   |
+LL |     &<A<T> as Foo<T>>::BAR
+   |     ^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This should be further improved by taking into account that promoteds are defined where they are used... but the `emit_note` logic doesn't know whether it is decorating a promoted. I'd leave that to a future PR.